sycl/source/detail/xpti_registry.hpp

```diff
@@ -0,0 +1,61 @@
+//==---------- xpti_registry.hpp ----- XPTI Stream Registry ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <string>
+#include <unordered_set>
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+// Include the headers necessary for emitting
+// traces using the trace framework
+#include "xpti_trace_framework.h"
+#endif
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+// We define a sycl stream name and this will be used by the instrumentation
+// framework
+inline constexpr const char *SYCL_STREAM_NAME = "sycl";
+// Stream name being used for traces generated from the SYCL plugin layer
+inline constexpr const char *SYCL_PICALL_STREAM_NAME = "sycl.pi";
+// Stream name being used for traces generated from PI calls. This stream
+// contains information about function arguments.
+inline constexpr const char *SYCL_PIDEBUGCALL_STREAM_NAME = "sycl.pi.debug";
```

Comment:
These were present in public headers (`include/` directory). Were these not used there?

Reply:
No, those were not used publicly, and I don't think it is a good idea to have these names in user-accessible headers. While user application can certainly use XPTI to produce even more perf data, those three streams are SYCL runtime-specific, and users should not be able to mess around with them.

Reply:
Agreed! The supporting documentation should give the developer trying to capture the traces from the streams to determine the stream name and what each stream provides.